src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitNotificationEventService.kt

```diff
@@ -109,7 +115,26 @@ class VisitNotificationEventService(
       val startDateTime = (if (LocalDate.now() > notificationDto.validFromDate) LocalDate.now() else notificationDto.validFromDate).atStartOfDay()
       val endDateTime = notificationDto.validToDate?.atTime(LocalTime.MAX)
       val affectedVisits = visitService.getFutureVisitsBy(notificationDto.prisonerNumber, prisonCode, startDateTime, endDateTime)
-      processVisitsWithNotifications(affectedVisits, PRISONER_RESTRICTION_CHANGE_EVENT)
+
+      processVisitsWithNotifications(affectedVisits, PRISONER_RESTRICTION_CHANGE_EVENT, null)
+    }
+  }
+
+  fun handlePrisonerAlertCreatedUpdatedNotification(notificationDto: PrisonerAlertCreatedUpdatedNotificationDto) {
+    LOG.debug("handlePrisonerAlertCreatedUpdated notification received")
+
+    val prisonerSupportedAlertCodes = PrisonerSupportedAlertCodeType.entries.map { it.name }.toSet()
+    val prisonerAlerts = notificationDto.alertsAdded.filter { code -> code in prisonerSupportedAlertCodes }
+
+    if (prisonerAlerts.isNotEmpty()) {
+      val prisonCode = prisonerService.getPrisonerPrisonCode(notificationDto.prisonerNumber)
+      val affectedVisits = visitService.getFutureVisitsBy(notificationDto.prisonerNumber, prisonCode)
+
+      val alertDescriptions = prisonerAlerts.map { alert ->
+        PrisonerSupportedAlertCodeType.entries.first { it.name == alert }.description
```

Comment:
should we add the PrisonerSupportedAlertCodeType - code and type as well? could add custom toString on PrisonerSupportedAlertCodeType if using all these fields.